nano/node/election.cpp

```diff
@@ -298,3 +298,22 @@ void nano::election::clear_blocks ()
 		}
 	}
 }
+
+void nano::election::insert_inactive_votes_cache ()
+{
+	auto winner_hash (status.winner->hash ());
+	auto reps (node.active.find_inactive_votes_cache (winner_hash));
+	for (auto & rep : reps)
+	{
+		auto last_vote_it (last_votes.find (rep));
+		if (last_vote_it == last_votes.end ())
+		{
+			last_votes[rep] = { std::chrono::steady_clock::time_point::min (), 0, winner_hash };
```

Comment:
Does this code need to be performant? If so you may consider using lower_bound/insert instead of operator[] (Effective STL - Item 24, Scott Meyers)

Reply:
Is it true for unordered_map as well?

Reply:
Ah no sorry thought I checked that, it only works with map. Actually because we don't care about updating (only inserting) the whole thing can be replace with an `emplace`, something like:
`last_votes.emplace (rep, vote_info{std::chrono::steady_clock::time_point::min (), 0, winner_hash});`
This only inserts if the key doesn't already exist:
https://en.cppreference.com/w/cpp/container/unordered_map/emplace

Reply:
Replaced